singleheader/amalgamate.py

```diff
@@ -75,10 +75,11 @@ def dofile(fid, prepath, filename):
     RELFILE = os.path.relpath(file, PROJECTPATH)
     # Last lines are always ignored. Files should end by an empty lines.
     print(f"/* begin file {RELFILE} */", file=fid)
+    includepattern = re.compile('^#include "(.*)"')
```

Comment:
This is a good pattern with pre-compilation, but it's actually unnecessary in small scripts. `re.search` (and all those other methods) has a global regex cache of the last 512 patterns used.

Reply:
This is informative, but I don't think you are saying that the script should be modified, right?

Reply:
Right, just a tidbit :)